referenceframe/frame_system.go

```diff
@@ -308,6 +309,48 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD in a framesystem and returns a map where all elements are the point representation
+// of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input) (map[string][]r3.Vector, error) {
```

Comment:
This should be broken into two functions.

Most of the internals here should go into something called `FrameSystemGeometries`, with the same parameters, but which returns an array of geometries.

`FrameSystemToPCD` should call that function to get the geometries, and then call `ToPoints` on each returned geometry to create the PCD to return.